### PR TITLE
ISSUE-16 - Further unit testing

### PR DESCRIPTION
**Pull request by jmcconnell26**

### CODE REVIEW

**Comment by anderejd** [comment]
Thanks for chipping away at the test suite 👍 

Are the formatting changes done by `cargo fmt`?

**Comment by jmcconnell26** [comment]
Hi,
No worries! I just want to be sure that the changes I make for removing the cargo crate don't break anything!
And yeah, the changes to the formatting are just `cargo fmt`.
Do you have a particular view on using it?

I was also wondering, would you mind if I raise a PR in https://github.com/rust-unofficial/awesome-rust mentioning `cargo-geiger`? I found it really useful when I was getting started in rust.

Thanks,
Josh

**Comment by anderejd** [comment]
> And yeah, the changes to the formatting are just cargo fmt.
Do you have a particular view on using it?

I love code formatting tools in general and `cargo fmt` / `rustfmt` is great!

> I was also wondering, would you mind if I raise a PR in https://github.com/rust-unofficial/awesome-rust mentioning cargo-geiger? I found it really useful when I was getting started in rust.

Sounds good to me 👍

**Comment by tarcieri** [comment]
@anderejd if you'd like I can add a CI check for rustfmt

**Comment by anderejd** [comment]
That would be perfect!

File: cargo-geiger/src/tree/traversal/dependency_node.rs [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+                    format!("test_name_{}", i),
+                    format!("1.2.{}", i).as_str(),
+                    source_id,
+                )
+                .unwrap(),
+            )
+        }
+
+        package_id_vec
+    }
```
Comment (anderejd) [inline cargo-geiger/src/tree/traversal/dependency_node.rs:231]: Minor, formatting: This function would be easier to read imho without any extra empty lines.

Comment (jmcconnell26) [reply cargo-geiger/src/tree/traversal/dependency_node.rs]: Good shout! Have pushed an update.

For the use of `cargo format` would it also be worth drafting a contributing guideline as well?

Comment (anderejd) [reply cargo-geiger/src/tree/traversal/dependency_node.rs]: Sure, a short contributing section in the readme file could be useful.

**Comment by anderejd** [comment]
Sorry for the conflict, I just pushed a `cargo fmt` to master.

**Comment by jmcconnell26** [comment]
No worries! Have merged

**Pull request merged**